state/application.go

```diff
@@ -290,7 +286,6 @@ func (a *Application) removeOps(asserts bson.D) ([]txn.Op, error) {
 		return nil, errors.Trace(err)
 	}
 	ops = append(ops, charmOps...)
-	ops = append(ops, finalAppCharmRemoveOps(name, curl)...)
```

Comment:
I don't know what 'finalAppCharmRemoveOps' are off hand. Is it the sort of thing that destroys a machine one the last unit on that machine is removed? That doesn't sound like something we want to drop, though I would think the tests should be testing that.